examples/IPC_Solver/genesis_ipc_motion_test.py

```diff
@@ -0,0 +1,242 @@
+            print(f"")
+            print(f"TOTAL  mom:  [{total_p[0]:9.5f}, {total_p[1]:9.5f}, {total_p[2]:9.5f}] kg·m/s")
+            print(f"|p_total|: {np.linalg.norm(total_p):.6f} kg·m/s")
+
+        scene.step()
+
+    # Convert lists to numpy arrays for plotting
+    time_history = np.array(time_history)
+    rigid_p_history = np.array(rigid_p_history)  # Shape: (n_steps, 3)
+    fem_p_history = np.array(fem_p_history)  # Shape: (n_steps, 3)
+    total_p_history = np.array(total_p_history)  # Shape: (n_steps, 3)
+    rigid_v_history = np.array(rigid_v_history)  # Shape: (n_steps, 3)
+    fem_v_history = np.array(fem_v_history)  # Shape: (n_steps, 3)
+
+    # Create plots
+    fig, axes = plt.subplots(2, 2, figsize=(14, 8))
+    fig.suptitle("Momentum Conservation Test (Zero Gravity)", fontsize=16)
+
+    # Plot 1: Linear Momentum X-component
+    ax = axes[0, 0]
+    ax.plot(time_history, rigid_p_history[:, 0], "r-", label="Rigid px", linewidth=2)
+    ax.plot(time_history, fem_p_history[:, 0], "b-", label="FEM px", linewidth=2)
+    ax.plot(time_history, total_p_history[:, 0], "k--", label="Total px", linewidth=2)
+    ax.axhline(y=4.0 * rigid_mass, color="g", linestyle=":", label=f"Expected: {4.0 * rigid_mass:.4f}")
+    ax.set_xlabel("Time (s)")
+    ax.set_ylabel("Momentum (kg·m/s)")
+    ax.set_title("Linear Momentum X-component")
+    ax.legend()
+    ax.grid(True, alpha=0.3)
+
+    # Plot 2: Total Momentum Magnitude
+    ax = axes[0, 1]
+    total_p_mag = np.linalg.norm(total_p_history, axis=1)
+    ax.plot(time_history, total_p_mag, "k-", linewidth=2, label="|p_total|")
+    ax.axhline(y=4.0 * rigid_mass, color="r", linestyle="--", label=f"Expected: {4.0 * rigid_mass:.4f}")
+    ax.set_xlabel("Time (s)")
+    ax.set_ylabel("|p| (kg·m/s)")
+    ax.set_title("Total Momentum Magnitude")
+    ax.legend()
+    ax.grid(True, alpha=0.3)
+
+    # Plot 3: Velocity X-component
+    ax = axes[1, 0]
+    ax.plot(time_history, rigid_v_history[:, 0], "r-", label="Rigid vx", linewidth=2)
+    ax.plot(time_history, fem_v_history[:, 0], "b-", label="FEM vx", linewidth=2)
+    ax.set_xlabel("Time (s)")
+    ax.set_ylabel("Velocity (m/s)")
+    ax.set_title("Velocity X-component")
+    ax.legend()
+    ax.grid(True, alpha=0.3)
+
+    # Plot 4: Momentum Conservation Error
+    ax = axes[1, 1]
+    expected_momentum = np.array([4.0 * rigid_mass, 0.0, 0.0])
+    momentum_error = np.linalg.norm(total_p_history - expected_momentum, axis=1)
+    ax.plot(time_history, momentum_error, "r-", linewidth=2)
+    ax.set_xlabel("Time (s)")
+    ax.set_ylabel("Error (kg·m/s)")
+    ax.set_title("Momentum Conservation Error")
+    ax.set_yscale("log")
+    ax.grid(True, alpha=0.3)
+
+    plt.tight_layout()
+    plt.savefig("momentum_conservation_test.png", dpi=150)
+    print(f"\n{'='*70}")
+    print("Plot saved to: momentum_conservation_test.png")
+    print(f"{'='*70}")
+    plt.show()
+
```

Comment:
It might be nice to add this tests to unit test and CI. But since right now libuipc is not on pypi and setting up libuipc environment in CI can take some work. We might wait till libuipc is available.